### DISCUSSION TITLE
JsPsych behavioural task

### DISCUSSION THREAD

**Discussion by Shirintaghian** [jspsych/jsPsych#2934] [Q&A]
Hello, 

Happy new year!

I am creating an experiment with JsPsych that’s based on recognition of facial emotions. So far I have 60 trials where an image is shown and then participant gets to pick a correct facial emotion that’s being present. However, I’ve encountered an issue where I need to check if the participant is actually present and doing the experiment, rather than picking random choices. To do that, I was hoping to end the timeline of trials if for example <10 choices are picked incorrectly in the experiment. But, I’m having issues with doing that. 
I was wondering if you’re able to help with this, I’ve tried to look through GitHub discussions and nothing really works. I’m attaching the code below. Please look into it if you have time, it would be greatly appreciated. Thank you. 

```js
 var correct_count = 0;

        var node = {
            timeline_variables: timeline_variables,
            timeline: [
            {
              type:jsPsychHtmlKeyboardResponse,
              stimulus: '<div style="font-size:60px;">+</div>',
              choices: "NO_KEYS",
              trial_duration:500,
            },
            {
              type:jsPsychImageKeyboardResponse,
              stimulus: 'group1/Unknown.png', 
              trial_duration:100
            },
            {
            type: jsPsychImageKeyboardResponse,
            stimulus: jsPsych.timelineVariable('v1'), 
            trial_duration: 100
            },
            {
              type:jsPsychImageKeyboardResponse,
              stimulus: 'group1/Unknown.png', 
              trial_duration:100
            },
            {
            type:jsPsychHtmlButtonResponse,
            stimulus: '<p>Which emotion among the two bottom did you notice in the image? </p>',
            choices: jsPsych.randomization.repeat([jsPsych.timelineVariable('v2'),jsPsych.timelineVariable('v3')],1),
            trial_duration:1500,
            data: [jsPsych.timelineVariable('v2'),jsPsych.timelineVariable('v3')],
            on_finish: function(data){
                 data.correct = jsPsych.timelineVariable('v3')
                 console.log(data.correct)
              if (data.response.resp == data.correct){
                correct_count++
                console.log(correct_count)
              }
            }
            },
            {
              type:jsPsychImageKeyboardResponse,
              stimulus: 'group1/Unknown.png', 
              trial_duration:2000
            }
          ],
          randomize_order: true,
        };

        var if_node = {
          timeline: [node],
          conditional_function: function(){
            console.log(correct_count)
            if (correct_count < 10){
              return true
            }
          }
        }
```

**Comment by jodeleeuw** [comment]
Hi @Shirintaghian,

You can use `jsPsych.endCurrentTimeline()` to end the timeline ([relevant docs here](https://www.jspsych.org/7.3/reference/jspsych/#jspsychendcurrenttimeline)). You could put this in the `on_finish` function of the trial where you are counting correct responses. 

If you need to count the number of correct responses over the last `n` trials you can use the features in `jsPsych.data` DataCollections. Here are [the docs for those features](https://www.jspsych.org/7.3/reference/jspsych-data/#datacollection).
